cmd/cloudtrail/write-events.go

```diff
@@ -0,0 +1,325 @@
+		Type           string `json:"type"`
+		SessionContext struct {
+			SessionIssuer struct {
+				Type     string `json:"type"`
+				UserName string `json:"userName"`
+				Arn      string `json:"arn"`
+			} `json:"sessionIssuer"`
+		} `json:"sessionContext"`
+	} `json:"userIdentity"`
+	EventRegion string `json:"awsRegion"`
+	EventId     string `json:"eventID"`
+}
+
+func newCmdWriteEvents() *cobra.Command {
+	ops := &LookupEventsOptions{}
+	listEventsCmd := &cobra.Command{
+		Use:   "write-events",
+		Short: "Prints out all cloudtrail write events to console",
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ops.run())
+		},
+	}
+	listEventsCmd.Flags().StringVarP(&ops.clusterID, "cluster-id", "C", "", "Cluster ID")
+	listEventsCmd.Flags().StringVarP(&ops.startTime, "since", "", "", "Start time of the lookup i.e (\"2h\" for starttime 2 hours ago")
+	listEventsCmd.Flags().BoolVarP(&ops.printEventUrl, "url", "u", false, "Generates Url link to cloud console cloudtrail event")
+	listEventsCmd.Flags().BoolVarP(&ops.printRawEvents, "raw-event", "r", false, "Prints the cloudtrail events to the console in raw json format")
+	listEventsCmd.Flags().BoolVarP(&ops.printAllEvents, "all", "A", false, "Prints all cloudtrail write events without filtering")
+	listEventsCmd.MarkFlagRequired("cluster-id")
+	return listEventsCmd
+}
+
+// parseDurationToUTC parses the given startTime string as a duration and subtracts it from the current UTC time.
+// It returns the resulting time and any parsing error encountered.
+func parseDurationToUTC(input string) (time.Time, error) {
+	duration, err := time.ParseDuration(input)
+	if err != nil {
+		return time.Time{}, err
+	}
+
+	return time.Now().UTC().Add(-duration), nil
+}
+
+// whoami retrieves caller identity information
+func whoami(stsClient sts.Client) (Arn string, AccountId string, err error) {
+	ctx := context.TODO()
+	callerIdentityOutput, err := stsClient.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{})
+	if err != nil {
+		return "", "", err
+	}
+
+	userArn, err := arn.Parse(*callerIdentityOutput.Arn)
+	if err != nil {
+		return "", "", err
+	}
+
+	return userArn.String(), userArn.AccountID, nil
+}
+
+// getWriteEvents retrieves cloudtrail events since the specified time
+// using the provided cloudtrail client and starttime from since flag.
+func getWriteEvents(since time.Time, cloudtailClient *cloudtrail.Client) ([]*cloudtrail.LookupEventsOutput, error) {
+	ctx := context.TODO()
+	starttime := since
+
+	allookupOutputs := []*cloudtrail.LookupEventsOutput{}
+	input := cloudtrail.LookupEventsInput{
+		StartTime: &starttime,
+		EndTime:   aws.Time(time.Now()),
+		LookupAttributes: []types.LookupAttribute{
+			{AttributeKey: "ReadOnly",
+				AttributeValue: aws.String("false")},
+		},
+	}
+
+	pages := 0
+	paginator := cloudtrail.LookupEventsPaginator{}
+	hasPages := paginator.HasMorePages()
+
+	for pageIter := 1; hasPages; pageIter++ {
+		pages += pageIter
+	}
```

Comment:
Is this an infinite loop? hasPages is a bool, if it's true it's always true and never gets updated here?